src/renderer/enterPassword.html

```diff
@@ -0,0 +1,155 @@
+              <h2>Enter password</h2>
+              <input
+                class="pw-input"
+                type="password"
+                id="enter-password-input"
+                placeholder="password"
+              />
+              <div class="warning" id="warning-1"></div>
+              <button style="margin-top: 5px" id="launch-btn" disabled>Launch</button>
+            </div>
+          </div>
+
+          <!-- Loading message -->
+          <div class="splash-loading-message hidden" id="activity">Loading...</div>
+        </div>
+        <div class="splash-license">Licensed under the Cryptographic Autonomy License v1.0</div>
+      </div>
+      <img
+        style="position: absolute; bottom: 5px; right: 10px; width: 300px"
+        src="img/powered_by_holochain.png"
+      />
+      <img
+        tabindex="0"
+        id="close-btn"
+        title="close"
+        style="cursor: pointer; position: absolute; top: 5px; right: 5px; width: 35px"
+        src="img/close.svg"
+      />
+    </div>
+    <script>
+      // TODO add event listener for holochain fatal panic case
+
+      let password = '';
+
+      (async () => {
+        const { productName, version } = await window.electronAPI.getNameAndVersion();
+        document.getElementById('productName').innerHTML = productName;
+        document.getElementById('version').innerHTML = `version ${version}`;
+      })();
+
+      const activitEl = document.getElementById('activity');
+      window.electronAPI.onProgressUpdate((event, msg) => {
+        console.log('status', msg);
+        activitEl.innerHTML = msg;
+      });
+
+      const closeBtn = document.getElementById('close-btn');
+      closeBtn.addEventListener('click', () => {
+        window.electronAPI.exit();
+      });
+      closeBtn.addEventListener('keypress', (e) => {
+        if (e.key === 'Enter' || e.key === ' ') {
+          window.electronAPI.exit();
+        }
+      });
+
+      const enterPwDiv = document.getElementById('enter-password-div');
+
+      const warningEl = document.getElementById('warning-1');
+
+      const enterPwInput = document.getElementById('enter-password-input');
+
+      enterPwInput.focus();
+
+      const handleLaunch = async () => {
+        enterPwDiv.classList.add('hidden');
+        activitEl.classList.remove('hidden');
+        try {
+          await window.electronAPI.launch({ type: 'user-provided', password });
+        } catch (e) {
+          console.error('Failed to launch: ', e);
+          activitEl.innerHTML = '';
+          activitEl.classList.add('hidden');
+          enterPwDiv.classList.remove('hidden');
+          if (e.toString().includes('Wrong password.')) {
+            password = '';
+            enterPwInput.value = '';
+            warningEl.innerHTML = 'Wrong Password.';
+            setTimeout(() => {
+              enterPwInput.focus();
+            });
+          } else {
+            // TODO add logic to offer exporting log files
```

Comment:
Is this meant to be handled in another PR?

Reply:
Yes I would do this as a separate PR. I made an issue for this and for handling holochain startup panics [here](https://github.com/holochain/kangaroo-electron/issues/21)